datafusion/sqllogictest/test_files/predicates.slt

```diff
@@ -725,3 +725,57 @@ AggregateExec: mode=SinglePartitioned, gby=[p_partkey@2 as p_partkey], aggr=[SUM
 --------CoalesceBatchesExec: target_batch_size=8192
 ----------RepartitionExec: partitioning=Hash([ps_partkey@0], 4), input_partitions=1
 ------------MemoryExec: partitions=1, partition_sizes=[1]
+
+# Inlist simplification
```

Comment:
I recommend one or two tests in slt, and then adding the other tests in https://github.com/apache/arrow-datafusion/blob/795c71f3e8249847593a58dafe578ffcbcd3e012/datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs#L1302-L1307 because:
1. It would keep the tests closer to the other simplification tests
2. It makes it easier to validate what code is causing the rewrite